data/docs/messaging-queues/kafka.mdx

```diff
@@ -415,9 +415,417 @@ Run the collector with the above configuration :
+      name: kafka_server_brokermetadatametrics_$1
+      type: GAUGE
+```
+
+Apply the ConfigMap:
+```bash
+kubectl apply -f kafka-metrics-config.yaml
+```
+
+### Step 4: Deploy OpenTelemetry Collector
+
+Install the SigNoz Kubernetes collector with automatic Strimzi discovery:
+
+```yaml
+global:
+  cloud: <aws, azure, gcp, others>
+  clusterName: signoz-demo-cluster
+  deploymentEnvironment: production
+otelCollectorEndpoint: <Collector Endpoint>
+otelInsecure: false
+signozApiKey: <API Key Here>
+presets:
+  otlpExporter:
+    enabled: true
+  loggingExporter:
+    enabled: false
+otelDeployment:
+  config:
+    receivers:
+      kafkametrics:
+        collection_interval: 10s
+        cluster_alias: signoz-demo-cluster1
+        brokers:
+          - signoz-demo-cluster1-kafka-bootstrap:9092
+        protocol_version: 3.0.0
+        scrapers:
+          - brokers
+          - topics
+          - consumers
+      prometheus:
+        config:
+          scrape_configs:
+            - job_name: 'strimzi-pods'
+              scrape_interval: 10s
+              kubernetes_sd_configs:
+              - role: pod
+              relabel_configs:
+              - source_labels: [__meta_kubernetes_pod_label_strimzi_io_kind]
+                regex: Kafka
+                action: keep
+```
+
+Deploy the collector:
+```bash
+helm install my-release signoz/k8s-infra -f ./signoz-k8.yml
+```
+
+### Step 5: Deploy Producer and Consumer Applications
+
+Deploy the Kafka producer and consumer applications with OpenTelemetry instrumentation:
+
+**Compile your Java producer and consumer applications:** Ensure your producer and consumer apps are compiled and ready to run.
+
+**Run Producer App with Java Agent:**
+
+```bash
+java -javaagent:/path/to/opentelemetry-javaagent.jar \
+     -Dotel.service.name=producer-svc \
+     -Dotel.traces.exporter=otlp \
+     -Dotel.metrics.exporter=otlp \
+     -Dotel.logs.exporter=otlp \
+     -jar /path/to/your/producer.jar
+```
+
+**Run Consumer App with Java Agent:**
+
+```bash
+java -javaagent:/path/to/opentelemetry-javaagent.jar \
+     -Dotel.service.name=consumer-svc \
+     -Dotel.traces.exporter=otlp \
+     -Dotel.metrics.exporter=otlp \
+     -Dotel.logs.exporter=otlp \
+     -Dotel.instrumentation.kafka.producer-propagation.enabled=true \
+     -Dotel.instrumentation.kafka.experimental-span-attributes=true \
+     -Dotel.instrumentation.kafka.metric-reporter.enabled=true \
+     -jar /path/to/your/consumer.jar
+```
+
+### Example: Deploying Sample Kafka Producer and Consumer Applications with OpenTelemetry Java Agent
+
+```yaml
+# Producer Deployment
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kafka-producer1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kafka-producer1
+  template:
+    metadata:
+      labels:
+        app: kafka-producer1
+    spec:
+      containers:
+      - name: producer
+        image: jaikanthjay46/kafka-consumer-test:latest
```

Comment:
@jaikanthjay46 can you please put a note mentioning that the user should change the image to their own docker image.